packages/php-wasm/universal/src/lib/urls.ts

```diff
@@ -59,3 +61,36 @@ export function ensurePathPrefix(path: string, prefix: string): string {
 	}
 	return prefix + path;
 }
+
+/**
+ * Returns a absolute URL for the provided URL.
+ *
+ * If the provided URL is relative, it will be prepended with the base URL.
+ * If a absolute URL is provided, it will return the provided URL and ignore
+ * the base URL.
+ *
+ * @param  url     - The URL to convert to an absolute URL.
+ * @param  baseUrl - The base URL to use to convert the relative URL to an absolute URL.
+ * @returns The absolute URL.
+ */
+export function toAbsoluteUrl(url: string, baseUrl: URL): string {
+	if (url.startsWith('http')) {
+		return url;
+	}
+
+	/**
+	 * Each Playground URL must have a scope to correctly resolve the current site.
+	 *
+	 * If a scope is provided in the relative URL, we need to preserve it.
+	 *
+	 * If the scope is not provided in the relative URL,
+	 * we need to use the base URL pathname.
+	 *
+	 * We include the full base URL pathname in case it has subfolders in addition to the scope.
+	 * Base URLs can have subfolders in multi-sites that use subfolders instead of subdomains.
+	 */
+	if (url.startsWith('/scope:')) {
+		return baseUrl.origin + url;
```

Comment:
This is a very loaded `if` that makes assumptions about Playground URL structure despite living in a function using a generic name `toAbsoluteUrl`. Let's find a name that covers the Playground-specific aspect of it.

Reply:
I see now. I wanted to make it generic but scopes are Playground specific.

Could you please take another look at the code now?

`toAbsoluteUrl` is now renamed to `appendPathnameToBaseUrlAndMaintainScope` and moved to `@php-wasm/scopes`.

I think that `@php-wasm/scopes` is a better location for it, but I'm not sure about the function name and comments. I probably changed it too many times and can't see it clearly anymore.